sql/core/src/main/scala/org/apache/spark/sql/jdbc/OracleDialect.scala

```diff
@@ -81,6 +81,10 @@ private case object OracleDialect extends JdbcDialect {
     case _ => None
   }
 
+  override def quoteIdentifier(colName: String): String = {
+    s""""${colName.stripPrefix("\"").stripSuffix("\"")}""""
```

Comment:
Is it widely used workaround? Manually stripping the quotes looks a bandaid fix

Reply:
I also think so. How about close this?

Reply:
Hmhmhmhm .. yea .. let's close it if there's no better way .. If this is used kind of widely, then it's okie. feel free to reopen when you have some time to take a look.

Reply:
Could you check whether we issue a right message? https://issues.apache.org/jira/browse/SPARK-20319 The error message in the JIRA is pretty confusing. 

We added the quotes automatically and thus it looks weird to our end users. We need to figure out how to resolve the issue.

Reply:
How about add `isSupportQuotedIdentifier` to `JdbcDialect`:
```scala
  def isSupportQuotedIdentifier: Boolean = true

  /**
   * Quotes the identifier. This is used to put quotes around the identifier in case the column
   * name is a reserved keyword, or in case it contains characters that require quotes (e.g. space).
   */
  def quoteIdentifier(colName: String): String = {
    if(!isSupportQuotedIdentifier) {
      assert(!colName.startsWith("\"") || !colName.endsWith("\""),
        "This dialect does not support quoted identifier.")
    }
    s""""$colName""""
  }
```
`DB2Dialect`, `OracleDialect` and `PostgresDialect` should override it and set it to `false`. `MySQLDialect` do not need  override it.

Reply:
See what we are doing in JDBCRelation's verifyAndGetNormalizedPartitionColumn. Basically, before we calling `quoteIdentifier `, we should already normalize the column names. Let us do the similar things in `getInsertStatement`?

Reply:
If so, we need verify both `getInsertStatement` and `createTable`?
https://github.com/apache/spark/blob/5f0ddd2d6e2fdebf549207bbc4b13ca709eee3c4/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala#L729-L746

Reply:
How about defining a helper method for name normalization by checking a name char-by-char like `UnresolvedAttribute.parseAttributeName`? If we have a new caller of `quoteIdentifier` in future, we need duplicate code for the normalization. I think that might be error-prone...

Reply:
yes. I think we can do it in both insert and create table code paths. Please try to dedup the code.